common/src/main/java/com/gregtechceu/gtceu/client/renderer/item/TagPrefixItemRenderer.java

```diff
@@ -1,90 +1,94 @@
+import com.gregtechceu.gtceu.utils.GradientUtil;
+import com.mojang.blaze3d.platform.NativeImage;
+import net.minecraft.client.Minecraft;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.data.models.model.DelegatedModel;
 import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.world.item.Item;
 
-import javax.annotation.Nullable;
-import java.util.HashMap;
-import java.util.function.Consumer;
-import java.util.function.Supplier;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.gregtechceu.gtceu.client.renderer.block.MaterialBlockRenderer.LAYER_2_SUFFIX;
 
 /**
  * @author KilaBash
  * @date 2023/2/16
  * @implNote TagPrefixItemRenderer
  */
-public class TagPrefixItemRenderer extends IModelRenderer {
-    private static final Table<MaterialIconType, MaterialIconSet, TagPrefixItemRenderer> MODELS = Tables.newCustomTable(new HashMap<>(), HashMap::new);
-
-    private final Supplier<ResourceLocation> modelLocationSupplier;
+public class TagPrefixItemRenderer {
+    private static final Set<TagPrefixItemRenderer> MODELS = new HashSet<>();
 
-    private TagPrefixItemRenderer(MaterialIconType type, MaterialIconSet iconSet) {
-        super(null);
-        this.modelLocationSupplier = Suppliers.memoize(() -> type.getItemModelPath(iconSet, true));
+    public static void create(Item item, MaterialIconType type, MaterialIconSet iconSet) {
+        MODELS.add(new TagPrefixItemRenderer(item, type, iconSet));
     }
 
-    public static TagPrefixItemRenderer getOrCreate(MaterialIconType type, MaterialIconSet iconSet) {
-        if (!MODELS.contains(type, iconSet)) {
-            MODELS.put(type, iconSet, new TagPrefixItemRenderer(type, iconSet));
+    public static void reinitModels() {
+        for (TagPrefixItemRenderer model : MODELS) {
+            ResourceLocation itemId = BuiltInRegistries.ITEM.getKey(model.item);
+            GTDynamicResourcePack.addItemModel(itemId, new DelegatedModel(model.type.getItemModelPath(model.iconSet, true)));
+            //ModelTemplates.FLAT_ITEM.create(GTDynamicResourcePack.getItemModelLocation(itemId), TextureMapping.layer0(itemId.withPrefix("item/")), GTDynamicResourcePack::addItemModel);
         }
-        return MODELS.get(type, iconSet);
     }
 
-    @Environment(EnvType.CLIENT)
-    @Override
-    protected UnbakedModel getModel() {
-        return ModelFactory.getUnBakedModel(modelLocationSupplier.get());
-    }
+    public static void initTextures() {
+        for (TagPrefixItemRenderer model : MODELS) {
+            ResourceLocation itemId = BuiltInRegistries.ITEM.getKey(model.item);
+
+            Resource file1 = Minecraft.getInstance().getResourceManager().getResource(GTDynamicResourcePack.getTextureLocation(null, model.type.getItemTexturePath(model.iconSet, true)/*.withSuffix("_layer1")*/)).orElse(null);
```

Comment:
I think resource loading could be extracted from here and `MaterialBlockRenderer` to some common utility

Reply:
well I'd say it's the same result and probably not that much of a longer line anyway in the end. Thought about it though.

Reply:
Ah sorry, my bad - I mean the entire resource loading block (probably until around line 81) looks like it could potentially be extracted ^^

Reply:
oh IC what you mean
that's the thing that is commented out in MixinHelpers and totally unused, but might prove useful later on.